src/NHapi.Base/Parser/MessageIterator.cs

```diff
@@ -182,6 +201,26 @@ public virtual void Reset()
         {
         }
 
+        private IStructureDefinition CouldBeNonGreedy()
```

Comment:
I know the naming of this method comes from [hapi](https://github.com/hapifhir/hapi-hl7v2/blob/3333e3aeae60afb7493f6570456e6280c0e16c0b/hapi-base/src/main/java/ca/uhn/hl7v2/parser/MessageIterator.java#L166) but it doesn't feel like a well named method, perhaps it should be called something in line with what the method does perhaps something like 
 `LocateNonGreedyStructure()` ? 
 
 what do you think?